packages/cursorless-engine/src/languages/TreeSitterQuery/QueryCapture.ts

```diff
@@ -6,13 +6,19 @@ import { Point } from "web-tree-sitter";
  * {@link MutableQueryCapture} to avoid using range/text and other mutable
  * parameters directly from the node.
  */
-export interface SimpleSyntaxNode {
+interface SimpleSyntaxNode {
   readonly id: number;
   readonly type: string;
+  readonly parent: SimpleSyntaxNode | null;
+  readonly children: Array<SimpleChildSyntaxNode>;
+}
+
+/**
+ * Add start and end position to the simple syntax node. Used by the `child-range!` predicate.
+ */
+interface SimpleChildSyntaxNode extends SimpleSyntaxNode {
```

Comment:
just to make sure I'm understanding, you've just cleaned it up so we don't need to use the `Omit`; the types are just as strict as before

Reply:
> just to make sure I'm understanding, you've just cleaned it up so we don't need to use the `Omit`; the types are just as strict as before

Exactly. Even a little bit more. Now we don't have start and and position on the parent for example. I was trying to make it more explicit that it's just for the children and just for this specific predicate we need to have these fields.